realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -2976,4 +2977,150 @@ public void clear_all() {
         assertEquals(0, realm.where(Cat.class).count());
         assertTrue(realm.isEmpty());
     }
+
+    @Test
+    public void waitForChange_emptyDataChange() throws InterruptedException {
+        final CountDownLatch bgRealmOpened = new CountDownLatch(1);
+        final CountDownLatch bgRealmClosed = new CountDownLatch(1);
+        final AtomicBoolean bgRealmResult = new AtomicBoolean(false);
+        final AtomicLong bgReamCount = new AtomicLong(0);
+
+        // wait in background
+        final CountDownLatch signalTestFinished = new CountDownLatch(1);
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Realm realm = Realm.getInstance(realmConfig);
+                bgRealmOpened.countDown();
+                bgRealmResult.set(realm.waitForChange());
+                bgReamCount.set(realm.where(AllTypes.class).count());
+                realm.close();
+                bgRealmClosed.countDown();
+            }
+        });
+        thread.start();
+
+        TestHelper.awaitOrFail(bgRealmOpened);
+        // it is necessary to give a time window for the background realm to wait on change.
+        // if we're to use wait/notify pair, we might actually miss notification from background and
+        // the foreground thread might wait indefinitely.
+        Thread.sleep(200);
```

Comment:
Tests that has `Thread.sleep(200);` without `stopWaitForChange()` now don't have any wait. For the ones with `stopWaitForChange()`, however, sleeping still remains as `waitForChange()` calls `enable_wait_for_change` before start waiting.